modules/tinymce/src/core/main/ts/undo/Setup.ts

```diff
@@ -114,6 +117,11 @@ export const registerEvents = (editor: Editor, undoManager: UndoManager, locks:
       undoManager.add({} as UndoLevel, e);
       isFirstTypedCharacter.set(true);
     }
+
+    const isMetaBackspace = Env.os.isMacOS() && e.metaKey && e.key === 'Backspace';
+    if (isMetaBackspace) {
+      undoManager.beforeChange();
+    }
```

Comment:
Since this affects more than just mac ctrl-backspace is also busted on linux/windows. So maybe change the code to this. And the ctrlKey should be able to be triggered in a test so then at least it's tested on windows but would have to be a webdriver test since agar can't simulate modern key events yet (should really fix that). The changelog would need to be updated as well. This will expand the fix to more than just what is in the ticket but since the bug is present on all operating systems it worth expanding it.

```suggestion
    const hasOnlyMetaOrCtrlModifier = Env.os.isMacOS() ? e.metaKey : e.ctrlKey && !e.altKey;
    if (hasOnlyMetaOrCtrlModifier) {
      undoManager.beforeChange();
    }
```